frontend/src/action/index.js

```diff
@@ -0,0 +1,12 @@
+
+
+// const API_URL = "http://localhost:3001";
+// export const getallProducts = (data) => async(dispatch) => {
+//     try{
+//         const res = await axios.get(`${API_URL}/products`);
```

Comment:
We are not going to use actions because we are not using redux. 

If we were action creators should be free of side effects like api calls. 

And this dispatch is empty which will cause unnecessary re-renders without changing state. 

It is much simpler to create a context provider using react context and update the context after the api is successful and we don't need an extra library just for state manipulation. 

Btw I'm not trying to be strict but with a lot of new people on the team we have to be considerate of the learning curve we add to the project.

Reply:
yes , deleted some unnecessory files and stuffs.